src/runtime/device_interface.cpp

```diff
@@ -54,6 +54,34 @@ WEAK int copy_to_host_already_locked(void *user_context, struct halide_buffer_t
 
 }}} // namespace Halide::Runtime::Internal
 
+namespace {
+
+__attribute__((always_inline))
+void debug_log_and_validate_buf(void * user_context, const halide_buffer_t &buf,
```

Comment:
I think this should return an error code, in case someone has overridden halide_error to not abort